indexer/index.hpp

```diff
@@ -209,13 +209,37 @@ class Index : public MwmSet
     ForEachInIntervals(implFunctor, covering::FullCover, m2::RectD::GetInfiniteRect(), scale);
   }
 
-  // "features" must be sorted using FeatureID::operator< as predicate
+  // "features" must be sorted using FeatureID::operator< as predicate.
   template <typename F>
   void ReadFeatures(F & f, vector<FeatureID> const & features) const
   {
-    size_t currentIndex = 0;
-    while (currentIndex < features.size())
-      currentIndex = ReadFeatureRange(f, features, currentIndex);
+    ASSERT_GREATER(features.size(), 0, ());
+    auto fidIter = features.begin();
+    auto const endIter = features.end();
+    while (fidIter != endIter)
+    {
+      MwmId const & id = fidIter->m_mwmId;
+      // Skip unregistered mwm files.
+      if (id.IsAlive())
```

Comment:
No need to pre-check that id is alive. Just create MwmHandle and check that it's alive.